common/scala/src/main/scala/whisk/common/tracing/TracingProvider.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.common.tracing
+
+import akka.actor.ActorSystem
+import whisk.common.TransactionId
+import whisk.spi.Spi
+
+/**
+  * SPI for providing Tracing implementations
+  */
+trait TracingProvider extends Spi{
```

Comment:
here's an example of an interceptor implemented with brave directly, albeit for a more sophisticated state model https://github.com/yschimke/oksocial/blob/master/src/main/java/com/baulsupp/oksocial/tracing/ZipkinTracingListener.java